python/ray/serve/_private/router.py

```diff
@@ -594,6 +599,17 @@ async def assign_request(
     ) -> ReplicaResult:
         """Assign a request to a replica and return the resulting object_ref."""
 
+        response_id = uuid.uuid4()
+        assign_request_task = asyncio.current_task()
+        ray.serve.context._add_request_pending_assignment(
+            request_meta.request_id, response_id, assign_request_task
```

Comment:
We should probably use `internal_request_id` for this. `request_id` can be set by the client from the header and can be repeated/ other unexpected behavior.

Reply:
nice thanks for the suggestion, changed to use internal request id!